src/pages/ContentScripts/features/oss-gpt/view.tsx

```diff
@@ -92,8 +92,51 @@ const View = ({ theme, currentRepo, currentDocsName }: Props): JSX.Element => {
     // Options for the observer (which mutations to observe)
     const config = { attributes: true, childList: true, subtree: true };
     // Callback function to execute when mutations are observed
+    function exists(selector: JQuery<HTMLElement>) {
+      return $(selector).length > 0;
+    }
     const callback: MutationCallback = (mutationList, observer) => {
-      if ($('section.emoji-mart').length > 0) {
+      const conversationContainer = $('.rcw-conversation-container');
+      const conversationResizer = $('.rcw-conversation-resizer');
+      if (!exists(conversationResizer) && exists(conversationContainer)) {
+        //we only add a resizer when there is no resizer and there is an active  conversation container
+        const newContainer = $('<div>');
+        newContainer.attr('class', 'rcw-conversation-resizer');
+        conversationContainer.prepend(newContainer);
+      } else {
+        // if resizer exits we only add listeners to it when resizable is true
+        let isDragging = false;
+        let initialMouseX: number;
+        let initialWidth: number;
+        if (exists(conversationResizer)) {
+          let handleMouseDown: (event: JQuery.MouseDownEvent) => void;
+          let handleMouseMove: (event: JQuery.MouseMoveEvent) => void;
+          let handleMouseUp: (event: JQuery.MouseUpEvent) => void;
+          handleMouseDown = (event: JQuery.MouseDownEvent) => {
+            isDragging = true;
+            initialMouseX = event.clientX;
+            initialWidth = parseInt(conversationContainer.css('width'), 10);
+          };
+          handleMouseMove = (event: JQuery.MouseMoveEvent) => {
+            if (isDragging) {
+              console.log(document);
+              const mouseX = event.clientX;
+              const widthDiff = mouseX - initialMouseX;
+              conversationContainer.css(
+                'width',
+                `${initialWidth - widthDiff}px`
+              );
+            }
+          };
```

Comment:
if you add `console.log(anything)` in `handleMouseMove`, you will find the times of this function being called is doubled every time users click and drag. That slows the page noticeably.

Reply:
I think this is because a new listener is added every time the useEffect that contains the callback function is triggered.